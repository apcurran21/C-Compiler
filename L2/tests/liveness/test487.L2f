(@meh
	0
	%memmy <- 69
	rdi <- %memmy
	rsi <- 3
	call allocate 2
	%thing <- rax
	rdi <- %thing
	call print 1
	return
)
