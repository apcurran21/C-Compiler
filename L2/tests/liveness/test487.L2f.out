(
(in
(r12 r13 r14 r15 rbp rbx)
(%memmy r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rbp rbx rdi rsi)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rbp rbx %thing)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rax rbp rbx)
)

(out
(%memmy r12 r13 r14 r15 rbp rbx)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rbp rbx rdi rsi)
(r12 r13 r14 r15 rax rbp rbx)
(r12 r13 r14 r15 rbp rbx %thing)
(r12 r13 r14 r15 rbp rbx rdi)
(r12 r13 r14 r15 rax rbp rbx)
()
)

)
